quadriga/src/main/java/edu/asu/diging/quadriga/config/SecurityContext.java

```diff
@@ -4,82 +4,107 @@
+import org.springframework.security.config.http.SessionCreationPolicy;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
-import org.springframework.security.web.authentication.www.BasicAuthenticationEntryPoint;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
 import org.springframework.security.web.util.matcher.RequestMatcher;
 
+import edu.asu.diging.quadriga.config.web.ApiAuthenticationDetailsSource;
+import edu.asu.diging.quadriga.config.web.CitesphereTokenFilter;
+import edu.asu.diging.quadriga.core.citesphere.impl.CitesphereConnectorImpl;
 import edu.asu.diging.simpleusers.core.service.SimpleUsersConstants;
 
 @Configuration
 @EnableWebSecurity
 public class SecurityContext extends WebSecurityConfigurerAdapter {
+    
+    @Configuration
+    @Order(2)
+    public static class WebSecurityConfig extends WebSecurityConfigurerAdapter {
 
-    @Override
-    public void configure(WebSecurity web) throws Exception {
-        web
-                // Spring Security ignores request to static resources such as CSS or JS
-                // files.
-                .ignoring().antMatchers("/static/**");
-    }
-
-    @Override
-    protected void configure(HttpSecurity http) throws Exception {
-        HeadersConfigurer<HttpSecurity> config = http.cors().and().antMatcher("**").csrf()
-                .requireCsrfProtectionMatcher(new RequestMatcher() {
-                    @Override
-                    public boolean matches(HttpServletRequest arg0) {
-                        // don't require CSRF for REST calls
-                        if (arg0.getRequestURI().indexOf("/api/") > -1) {
-                            return false;
-                        }
-                        if (arg0.getRequestURI().indexOf("/rest/") > -1) {
-                            return false;
-                        }
-                        if (arg0.getMethod().equals("GET")) {
-                            return false;
+        @Override
+        public void configure(WebSecurity web) throws Exception {
+            web
+                    // Spring Security ignores request to static resources such as CSS or JS
+                    // files.
+                    .ignoring().antMatchers("/static/**");
+        }
+    
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            HeadersConfigurer<HttpSecurity> config = http.cors().and().antMatcher("**").csrf()
+                    .requireCsrfProtectionMatcher(new RequestMatcher() {
+                        @Override
+                        public boolean matches(HttpServletRequest arg0) {
+                            // don't require CSRF for REST calls
+                            if (arg0.getRequestURI().indexOf("/api/") > -1) {
+                                return false;
+                            }
+                            if (arg0.getRequestURI().indexOf("/rest/") > -1) {
```

Comment:
we don't have a rest endpoint

Reply:
I have not done any changes here, this was already there
I just added the config code in a separate class as per the files security config design
Should I remove the /rest if condition?

Reply:
yes, unless we have any controllers that use that prefix (which I'm pretty sure we don't have).